control-plane-operator/main.go

```diff
@@ -403,6 +411,8 @@ func NewStartCommand() *cobra.Command {
 			DefaultIngressDomain:          defaultIngressDomain,
 			MetricsSet:                    metricsSet,
 			NameServerIP:                  nameServerIP,
+			RHOBSMonitoring:               rhobsMonitoring,
+			IsManagementClusterOpenShift:  isManagementClusterOpenShift,
```

Comment:
why do we need `IsManagementClusterOpenShift`?

Same thinking than https://github.com/openshift/hypershift/pull/2807/files#r1406035100 
Can we please make this RHOBSMonitoring input explicit? e.g. needsManagementMetricsAccess or similar?
Then that new explicit being enable here depends on its dedicated input flag AND RHOBSMonitoring being disable.

Reply:
> why do we need IsManagementClusterOpenShift?

The PR aims the hosted CVO at the `thanos-querier` service in the `openshift-monitoring` namespace. This service provides the HCP metrics, but the service is present and contains the HCP metrics only in self-managed HyperShift running on an OpenShift management cluster. Thus the variable `IsManagementClusterOpenShift` is supposed to help aim the CVO at the mentioned service only when it is present (`IsManagementClusterOpenShift` == `true`) and contains the metrics (`RHOBSMonitoring` == `false`) to not point the hosted CVO at a non-existing service; for example, in case the HyperShift is running on a vanilla Kubernetes. 

We can omit this variable and leave a note in the description for the `EnableCVOManagementClusterMetricsAccess` flag including information like "_this will enable evaluation of conditional updates of hosted control planes (supported for OpenShift management clusters at the moment)_".

What do you think, @enxebre ?

---

> Then that new explicit input being enabled here depends on its dedicated input flag AND RHOBSMonitoring being disable.

Should then the new explicit input here `needsManagementMetricsAccess` depend only on its dedicated input flag whose value is propagated from `EnableCVOManagementClusterMetricsAccess` flag when installing HyperShift? A check for `EnableCVOManagementClusterMetricsAccess == true && RHOBSMonitoring == true` could then be present when installing HyperShift, and passing the condition would fail the install. Rather than dynamically modifying the value of `needsManagementMetricsAccess` inside the CPO.

Reply:
The `IsManagementClusterOpenShift` variable also helps to decide whether to create the API objects (NetworkPolicy, Role) when they are needed to aim the hosted CVO at the `thanos-querier.openshift-monitoring.svc` service.

Potentially we could remove the duplicating check for `if IsManagementClusterOpenShift && !RHOBSMonitoring` and only leave a check for something like `if needsManagementMetricsAccess`. The value of which is propagated from HyperShift install and cannot be set when `--rhobs-monitoring` is set.